src/clients/asset-client.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const Client = require('./base-client')
+const simpleFetch = require('../utils/simple-fetch')
+const mime = require('mime-types')
+
+class AssetClient {
+  constructor (url) {
+    this.client = new Client({
+      name: 'AssetClient',
+      url: url,
+      contentType: 'application/vnd.api+json'
+    })
+  }
+
+  create (params) {
+    return this.__createOrUpdate(params)
+  }
+
+  destroy (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets/${id}`, {
+      method: 'DELETE'
+    })
+  }
+
+  index () {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets`)
+  }
+
+  show (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets/${id}`)
+  }
+
+  update (params) {
+    return this.__createOrUpdate(params)
+  }
+
+  download (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}`)
+  }
+
+  display (id) {
```

Comment:
I'd probably prefer to use the `links.download` and `links.inline` (or whatever I named them) in the `links` attribute of the JSON object. Mostly because that means we wouldn't have to change the client, if we ever decide that we don't want to provide those links under the `api.bundesimmobilien.de` subdomain anymore. As we'd have to touch the API project anyway, we might also change the links there.

_Disclaimer:_ I am a fan of using links in API responses. Therefore, I might actually be a bit biased. 🙄

Reply:
I'm not sure I understand cause these are those links. I assumed `baseUrl` is `api.bundesimmobilien.de/asset-service` - ref: https://github.com/infopark-customers/bima-serverless/pull/1